### DISCUSSION TITLE
Command line to change the RGB numbers of images

### DISCUSSION THREAD

**Discussion by wisemike2** [ImageMagick/ImageMagick#4711] [Help]
I am trying to convert **all images** in a folder and subfolders to rgb: **(R:242 - G:235 - B: 190)**, but I can't seem to find the right command for that. All I found was this command that darkens the image, but when I changed gray47 to the numbers I want the output was black:

```
   convert \
       input.jpg \
      -channel red   -fx 'r - gray47' \
      -channel green -fx 'g - gray47' \
      -channel blue  -fx 'b - gray47' \
       output.jpg
```

ImageMagick version 6.8.9-9 Q16 x86_64

**Comment by snibgo** [comment]
Your "-fx" expressions subtract colours. The first one takes the current value in the "r" channel of the image, subtracts the "r" channel of "gray47", and writes the result to the "r" channel of the image.

But is that what you want to do?

You seem to want to change all pixels in the image to a constant colour. For that, "-colorize" is the obvious operation.
```
convert input.jpg -fill gray47 -colorize 100 output.jpg
```
... or:
```
convert input.jpg -fill rgb(242,235,190) -colorize 100 output.jpg
```
